### PR TITLE
add unit test : Requester and Responder set HANDSHAKE_IN_THE_CLEAR_CAP to 0.

### PR DESCRIPTION
**Pull request by Xiaohanjlll**
Fix: #2774

### CODE REVIEW

**Comment by jyao1** [comment]
need add test for test_spdm_requester as well.

File: unit_test/test_spdm_requester/finish.c [hunk 1]
```diff
@@ -3682,6 +3812,209 @@ void libspdm_test_requester_finish_case23(void **state)
...
+
+    assert_memory_equal((const void *)secured_message_context->master_secret.master_secret,
+                        (const void *)m_libspdm_zero_buffer, sizeof(m_libspdm_zero_buffer));
+    free(data);
+}
+
+/**
+ * Test 25: Set HANDSHAKE_IN_THE_CLEAR_CAP to 0 , The ResponderVerifyData field is present.
+ * Expected behavior: client returns a Status of RETURN_DEVICE_ERROR.
+ **/
```
Comment (Zhiqiang520) [inline unit_test/test_spdm_requester/finish.c:3925]: Here `RETURN_DEVICE_ERROR` is not aligned with `LIBSPDM_STATUS_INVALID_MSG_SIZE` at line 4014

Comment (Xiaohanjlll) [reply unit_test/test_spdm_requester/finish.c:3925]: Thanks , I have modified this place

**Comment by Xiaohanjlll** [comment]
> need add test for test_spdm_requester as well.

Thanks, Has added the test for test_spdm_requester

**Pull request merged**